core/templates/services/keyboard-shortcut.service.ts

```diff
@@ -19,6 +19,9 @@ import 'mousetrap';
 
 import { Injectable } from '@angular/core';
 import { downgradeInjectable } from '@angular/upgrade/static';
+import { WindowRef } from 'services/contextual/window-ref.service.ts';
+
+const windowRef = new WindowRef();
```

Comment:
Hi @jimbyo, can you use the angular's dependency injection mechanism here. All this means is you have to provide the class name and you will get a class instance.
Instead of having to instantiate yourself, you can make angular do it by providing this in the constructor. Something like:
```
constructor(private windowRef: WindowRef){}
```
This constructor will come at line number 30 (After `export class KeyboardShortcutService` and before `bindExplorationPlayerShortcuts() {`).
 Then you can access the windowRef by using the "this" keyword. So something like
```
windowRef.nativeWindow.location.href = href;
```
 will become:
```
this.windowRef.nativeWindow.location.href = href;
```

Reply:
Hey thanks for the suggestion. I actually originally did this I believe, I followed url.service.ts way of injecting the WindowRef, but it made it so that the keyboard shortcuts would no redirect to a new page. So I switched to doing it the current way

Reply:
@srijanreddy98, Thanks for helping out! 

@jimbyo, Do let us know if you need any further help. Thanks!